src/Builder/Encoder/FieldPathEncoder.php

```diff
@@ -0,0 +1,31 @@
+<?php
+
+declare(strict_types=1);
+
+namespace MongoDB\Builder\Encoder;
+
+use MongoDB\Builder\Type\FieldPathInterface;
+use MongoDB\Codec\EncodeIfSupported;
+use MongoDB\Exception\UnsupportedValueException;
+
+/** @template-extends AbstractExpressionEncoder<string, FieldPathInterface> */
+class FieldPathEncoder extends AbstractExpressionEncoder
+{
+    /** @template-use EncodeIfSupported<string, FieldPathInterface> */
+    use EncodeIfSupported;
+
+    public function canEncode(mixed $value): bool
+    {
+        return $value instanceof FieldPathInterface;
+    }
+
+    public function encode(mixed $value): string
+    {
+        if (! $this->canEncode($value)) {
+            throw UnsupportedValueException::invalidEncodableValue($value);
+        }
+
+        // TODO: needs method because of interface
```

Comment:
Noted that this was extracted from the inline logic in BuilderEncode. Should this TODO be ticketed?

Reply:
I'll defer to @GromNaN on this since this may involve more work but also isn't bad to cause a fatal error in case we forget it somewhere.

Reply:
Created PHPLIB-1379 to track this.